### PR TITLE
grt: skip layers out of min/max layer range

### PR DESCRIPTION
**Pull request by eder-matheus**
Fixes https://github.com/The-OpenROAD-Project/OpenROAD/issues/4253

### CODE REVIEW

File: src/grt/src/fastroute/src/FastRoute.cpp [hunk 1]
```diff
@@ -818,8 +819,8 @@ void FastRouteCore::updateDbCongestion()
   db_gcell->addGridPatternX(x_corner_, x_grid_, tile_size_);
   db_gcell->addGridPatternY(y_corner_, y_grid_, tile_size_);
   auto db_tech = db_->getTech();
-  for (int k = 0; k < num_layers_; k++) {
-    auto layer = db_tech->findRoutingLayer(k + 1);
+  for (int k = min_routing_layer - 1; k < max_routing_layer - 1; k++) {
```
Comment (maliberty) [inline src/grt/src/fastroute/src/FastRoute.cpp:822]: why min_routing_layer -1 ?

Comment (eder-matheus) [reply src/grt/src/fastroute/src/FastRoute.cpp:822]: In the GlobalRoute class, the min layer starts in 1. Inside FastRouteCore, the min layer is zero. So, we need the `-1` here to correctly access the vectors with k.

**Review by maliberty — APPROVED** [review]

**Comment by maliberty** [comment]
Please look at the CI failures

File: src/grt/src/fastroute/src/FastRoute.cpp (hunk 1 above)
Comment (eder-matheus) [reply src/grt/src/fastroute/src/FastRoute.cpp:822]: It should be fine with this. The changes in #4263 don't change the "global" layer range. It just makes the specific net layer ranges between FastRouteCore and GlobalRoute classes consistent.

Comment (maliberty) [reply src/grt/src/fastroute/src/FastRoute.cpp:822]: We should put it on the list to resolve this difference in indexing as it seems to be a bug source.

Comment (eder-matheus) [reply src/grt/src/fastroute/src/FastRoute.cpp:822]: I can create a GH issue to keep track of it.

Comment (eder-matheus) [reply src/grt/src/fastroute/src/FastRoute.cpp:822]: @maliberty FYI: https://github.com/The-OpenROAD-Project/OpenROAD/issues/4266

**Pull request merged**